tool/osl_pluto.c

```diff
@@ -1310,22 +1321,32 @@ static void compute_deps_osl(osl_scop_p scop, PlutoProg *prog,
 
         char name[25];
 
+        int nb_parameters = 0;
+        int nb_iterators = 0;
+        int nb_scattdims = 0;
+        int nb_localdims = 0;
+        int array_id = 0;
```

Comment:
All these variables should ideally be `unsigned`. I understand that the previous code itself wasn't proper here. We can fix it on this update for this part touched.

Reply:
I don't think they should. Those variables are set later with the call of `osl_relation_get_attributes` which prototype require `int *`: `void osl_relation_get_attributes(const osl_relation_t*, int*, int*, int*, int*);`
They can take the value `OSL_UNDEFINED` (which is -1) if the given relation is NULL. 
But OSL_UNDEFINED should be a better init value than 0.